### ISSUE TITLE
Allow bootstrap.servers to be provided via a PasswordProvider for Kafka Ingestion

### ISSUE THREAD

**Issue by JulianJaffePinterest** [apache/incubator-druid#8685]
### Description

Currently, the `consumerProperties` map in a KafkaSupervisorIOConfig can use PasswordProviders for the `keystore`, `truststore` and `key` properties, but `bootstrap.servers` must be a hard-coded String. Although `bootstrap.servers` is not a password, it would be helpful if the value for this property could be provided via a similar mechanism to a PasswordProvider. This would allow SupervisorSpecs to be robust to changes in the Kafka bootstrap servers in the event of a cluster migration or the like.

### Motivation

Currently, if the bootstrap servers for a Kafka cluster change, running supervisor specs that refer to those bootstrap servers must be manually killed and relaunched with references to the new bootstrap servers. If a cluster operator has access to a separate system that can provide updated bootstrap server information, these manual operations can be avoided by borrowing the concept of a "Provider" from the PasswordProvider.

**Comment by gianm** [comment]
Maybe it'd make sense to allow this for _any_ property?

**Comment by JulianJaffePinterest** [comment]
I agree that allowing most properties to be specified via a Provider makes sense. I think we'd have to figure out a good solution to #6666 or else users could encounter some very hard to debug Heisenbug-style errors though.
